### ISSUE TITLE
pgmoneta_ext_promote()

### ISSUE THREAD

**Issue by jesperpedersen** [pgmoneta/pgmoneta_ext#26]
Add a function that basically calls

```
SELECT * FROM pg_promote()
```

-- requires `SUPERUSER` -- and return the `true`/`false` value

**Comment by jesperpedersen** [comment]
@GuChad369 Can you take this one ?

**Comment by jesperpedersen** [comment]
* https://www.postgresql.org/docs/17/spi.html
* https://www.postgresql.org/docs/17/spi-examples.html

**Comment by jesperpedersen** [comment]
Any way to do it is ok

**Comment by GuChad369** [comment]
sure

**Comment by GuChad369** [comment]
The `pg_promote()` function can only be executed when the server is in recovery mode. Do we need to verify if the server is in recovery mode and return an appropriate message if it is not.

**Comment by jesperpedersen** [comment]
Yes, you can do that -- a simple `bool` back and forth is ok

**Comment by jesperpedersen** [comment]
I mean, the message can be displayed in the client calling the function

**Comment by GuChad369** [comment]
In PostgreSQL, a warning message is already displayed if the server is not in recovery mode. Below is an example of a custom extension function call and its direct output.

```sql
postgres=# SELECT pgmoneta_ext_promote();
ERROR:  recovery is not in progress
HINT:  Recovery control functions can only be executed during recovery.
postgres=# SELECT * FROM pg_promote();
ERROR:  recovery is not in progress
HINT:  Recovery control functions can only be executed during recovery.
```

**Comment by jesperpedersen** [comment]
Yeah, that is enough for debugging - the client for the extension just need to know `false` or `true`
